# Whether the client has subscribed a term deposit or not 


In [2]:
import pandas as pd

In [3]:
bank= pd.read_csv('bank-full.csv', sep=';')
bank

age           job   marital  education default  balance housing loan  \
0       58    management   married   tertiary      no     2143     yes   no   
1       44    technician    single  secondary      no       29     yes   no   
2       33  entrepreneur   married  secondary      no        2     yes  yes   
3       47   blue-collar   married    unknown      no     1506     yes   no   
4       33       unknown    single    unknown      no        1      no   no   
...    ...           ...       ...        ...     ...      ...     ...  ...   
45206   51    technician   married   tertiary      no      825      no   no   
45207   71       retired  divorced    primary      no     1729      no   no   
45208   72       retired   married  secondary      no     5715      no   no   
45209   57   blue-collar   married  secondary      no      668      no   no   
45210   37  entrepreneur   married  secondary      no     2971      no   no   

         contact  day month  duration  campaign  pdays  previous poutcome    y  
0        unknown    5   may       261         1     -1         0  unknown   no  
1        unknown    5   may       151         1     -1         0  unknown   no  
2        unknown    5   may        76         1     -1         0  unknown   no  
3        unknown    5   may        92         1     -1         0  unknown   no  
4        unknown    5   may       198         1     -1         0  unknown   no  
...          ...  ...   ...       ...       ...    ...       ...      ...  ...  
45206   cellular   17   nov       977         3     -1         0  unknown  yes  
45207   cellular   17   nov       456         2     -1         0  unknown  yes  
45208   cellular   17   nov      1127         5    184         3  success  yes  
45209  telephone   17   nov       508         4     -1         0  unknown   no  
45210   cellular   17   nov       361         2    188        11    other   no  

[45211 rows x 17 columns]

In [4]:
import pandas_profiling as pp

In [5]:
Bank=pp.ProfileReport(bank)
Bank.to_file(output_file="Bank.html")

In [6]:
bank.describe()

age        balance           day      duration      campaign  \
count  45211.000000   45211.000000  45211.000000  45211.000000  45211.000000   
mean      40.936210    1362.272058     15.806419    258.163080      2.763841   
std       10.618762    3044.765829      8.322476    257.527812      3.098021   
min       18.000000   -8019.000000      1.000000      0.000000      1.000000   
25%       33.000000      72.000000      8.000000    103.000000      1.000000   
50%       39.000000     448.000000     16.000000    180.000000      2.000000   
75%       48.000000    1428.000000     21.000000    319.000000      3.000000   
max       95.000000  102127.000000     31.000000   4918.000000     63.000000   

              pdays      previous  
count  45211.000000  45211.000000  
mean      40.197828      0.580323  
std      100.128746      2.303441  
min       -1.000000      0.000000  
25%       -1.000000      0.000000  
50%       -1.000000      0.000000  
75%       -1.000000      0.000000  
max      871.000000    275.000000

In [11]:
bank1=pd.get_dummies(bank,columns=['housing','loan','y','default','job','marital','education','contact','poutcome','month'])
bank1

age  balance  day  duration  campaign  pdays  previous  housing_no  \
0       58     2143    5       261         1     -1         0           0   
1       44       29    5       151         1     -1         0           0   
2       33        2    5        76         1     -1         0           0   
3       47     1506    5        92         1     -1         0           0   
4       33        1    5       198         1     -1         0           1   
...    ...      ...  ...       ...       ...    ...       ...         ...   
45206   51      825   17       977         3     -1         0           1   
45207   71     1729   17       456         2     -1         0           1   
45208   72     5715   17      1127         5    184         3           1   
45209   57      668   17       508         4     -1         0           1   
45210   37     2971   17       361         2    188        11           1   

       housing_yes  loan_no  ...  month_dec  month_feb  month_jan  month_jul  \
0                1        1  ...          0          0          0          0   
1                1        1  ...          0          0          0          0   
2                1        0  ...          0          0          0          0   
3                1        1  ...          0          0          0          0   
4                0        1  ...          0          0          0          0   
...            ...      ...  ...        ...        ...        ...        ...   
45206            0        1  ...          0          0          0          0   
45207            0        1  ...          0          0          0          0   
45208            0        1  ...          0          0          0          0   
45209            0        1  ...          0          0          0          0   
45210            0        1  ...          0          0          0          0   

       month_jun  month_mar  month_may  month_nov  month_oct  month_sep  
0              0          0          1          0          0          0  
1              0          0          1          0          0          0  
2              0          0          1          0          0          0  
3              0          0          1          0          0          0  
4              0          0          1          0          0          0  
...          ...        ...        ...        ...        ...        ...  
45206          0          0          0          1          0          0  
45207          0          0          0          1          0          0  
45208          0          0          0          1          0          0  
45209          0          0          0          1          0          0  
45210          0          0          0          1          0          0  

[45211 rows x 53 columns]

In [13]:
bank2 =bank1.drop(['housing_no','loan_no','y_no','default_no'],axis=1)
bank2

age  balance  day  duration  campaign  pdays  previous  housing_yes  \
0       58     2143    5       261         1     -1         0            1   
1       44       29    5       151         1     -1         0            1   
2       33        2    5        76         1     -1         0            1   
3       47     1506    5        92         1     -1         0            1   
4       33        1    5       198         1     -1         0            0   
...    ...      ...  ...       ...       ...    ...       ...          ...   
45206   51      825   17       977         3     -1         0            0   
45207   71     1729   17       456         2     -1         0            0   
45208   72     5715   17      1127         5    184         3            0   
45209   57      668   17       508         4     -1         0            0   
45210   37     2971   17       361         2    188        11            0   

       loan_yes  y_yes  ...  month_dec  month_feb  month_jan  month_jul  \
0             0      0  ...          0          0          0          0   
1             0      0  ...          0          0          0          0   
2             1      0  ...          0          0          0          0   
3             0      0  ...          0          0          0          0   
4             0      0  ...          0          0          0          0   
...         ...    ...  ...        ...        ...        ...        ...   
45206         0      1  ...          0          0          0          0   
45207         0      1  ...          0          0          0          0   
45208         0      1  ...          0          0          0          0   
45209         0      0  ...          0          0          0          0   
45210         0      0  ...          0          0          0          0   

       month_jun  month_mar  month_may  month_nov  month_oct  month_sep  
0              0          0          1          0          0          0  
1              0          0          1          0          0          0  
2              0          0          1          0          0          0  
3              0          0          1          0          0          0  
4              0          0          1          0          0          0  
...          ...        ...        ...        ...        ...        ...  
45206          0          0          0          1          0          0  
45207          0          0          0          1          0          0  
45208          0          0          0          1          0          0  
45209          0          0          0          1          0          0  
45210          0          0          0          1          0          0  

[45211 rows x 49 columns]

In [14]:
x=bank2.drop('y_yes',axis=1)

In [15]:
y=bank2['y_yes']

In [16]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test =  train_test_split(x,y,test_size = 0.25,random_state=42)

In [17]:
from sklearn.linear_model import LogisticRegression
model =  LogisticRegression(fit_intercept = True)

model.fit(x_train,y_train)


C:\Users\PANCHI\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:762: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [18]:
ypred=model.predict(x_test)
ypred

array([0, 0, 0, ..., 0, 0, 0], dtype=uint8)

In [20]:
x_test.shape

(11303, 48)

In [21]:
model.score(x_test,y_test)

0.8863133681323543

As we can see accuracy of our model is 0.8863 , we can predict whether the client has subscribed a term deposit or not 